### ISSUE TITLE
Crash/OOM after wakeup from suspend

### ISSUE THREAD

**Issue by Corwitep** [floccusaddon/floccus#990]
### Describe the bug
After Wakeup from Suspend Firefox with floccus consumes all available memory and is eventually killed by the oom killer. A similiar behaviour (excessive memory consumption) can be observed with chrome based browser too, but on those the memory consumption goes down after several seconds.

### To Reproduce
1. Start Firefox with Floccus
2. Put the computer into suspend
3. Wake it up
4. See error description

### Expected behavior
No crash

### Desktop
(please complete the following information)
 - OS: Linux (Manjaro/current with XFCE and Ubuntu 21.04 with XFCE tested)
 - Browser: Firefox, observed with chrome browsers (chrome, chromium, vivaldi) too
 - Browser Version: Current
 - Floccus version: 4.8.3
 - Floccus sync method: nextcloud

### Server
(please complete the following information)
 - OS: Linux (Ubuntu 20.04)
 - Nextcloud version: 22.2.0
 - Bookmarks app version: 10.0.0

**Comment by mnalis** [comment]
Do you have multiple syncs set up? If you do, does the same issue happens if you manually try to run all of them at the same time?

This is just a guess, but it so, it might be related that after waking up Floccus sees that the time has passed, and tries to run all of them at once...

**Comment by Corwitep** [comment]
No, only a single sync to a nextcloud server in the local network is configured. The issue does not happen with a manual sync, I've not found a way to reproduce it without putting the machine into suspend.
I had the suspicion too that floccus might try to do do _many_ syncs at once after the wakeup, because I observed that the problem does not seem to occur if the suspend was only for a very short time (no more than several seconds), but it definitely happens after a few minutes of suspend and it's unlikely any relevant number of sync operations could have been queued up in that timespan.
If neccessary I can try for example to strace the firefox process or provide debug logs (which one?).

**Comment by mnalis** [comment]
@Corwitep It would be good if you can try following things to narrow down where the problem might be:

- how often do you have floccus setup to auto-sync?  
- do you perhaps know if your system is using suspend-to-memory or suspend-to-disk (AKA "hibernate")? If it uses suspend-to-disk, can you change it to suspend-to-memory and see if the problem still happens?
- if you disable floccus extension completely (in `Tools` / `Add ons` / `Extensions`), does it still crash after suspend?
- if you enable floccus extension, but disable auto-sync for your account, does it still crash after suspend?
- could you capture output of following commands `free; ps aux|grep -i firefox; dmesg|grep -i firefox` at four different times:
    - right before you suspend
    - right after you resume
    - some time later after resume, but before  firefox crashes (this is finicky to get timing right, ideally it would be right before firefox crashes, or at least close to that time)
    - right after firefox has crashed
- instead of hibernating, with floccus and its auto-sync enabled, try to stop only firefox - eg. start Firefox from terminal with `firefox` CLI command instead of via GUI shortcut, and then pressing `ctrl-z` to suspend it (firefox window should become completely unresponsive, but you can use other windows), and after dozen minutes unsuspend it by typing `fg` in that same terminal. Does the crash/OOM still happens?
